flatpak-import.py

```diff
@@ -0,0 +1,212 @@
+        # checksum format invalid
+        exit(1)
+
+    metadata_variant = delta.get_child_value(0)
+
+    commit = OSTree.checksum_from_bytes_v(checksum_variant)
+
+    ref = metadata_variant['ref']
+
+    if 'flatpak' in metadata_variant.keys():
+        flatpak = metadata_variant['flatpak']
+    else:
+        flatpak = ''
+
+    if 'origin' in metadata_variant.keys():
+        origin = metadata_variant['origin']
+    else:
+        origin = ''
+
+    if 'runtime-repo' in metadata_variant.keys():
+        runtime_repo = metadata_variant['runtime-repo']
+    else:
+        runtime_repo = ''
+
+    if 'metadata' in metadata_variant.keys():
+        app_metadata = metadata_variant['metadata']
+    else:
+        app_metadata = ''
+
+    if 'gpg-keys' in metadata_variant.keys():
+        gpg_keys = metadata_variant['gpg-keys']
+    else:
+        gpg_keys = ''
+
+    logging.debug("commit: " + commit)
+    logging.debug("flatpak: " + commit)
+    logging.debug("ref: " + ref)
+    logging.debug("origin: " + origin)
+    logging.debug("runtime_repo: " + runtime_repo)
+    logging.debug("app_metadata:\n---\n" + app_metadata + "---")
+    logging.debug("gpg_keys: " + gpg_keys)
+
+    logging.debug("metadata keys: " + str(metadata_variant.keys()))
+
+
+    # open repository
+
+    repo_file = Gio.File.new_for_path(args.repo)
+    repo = OSTree.Repo(path=repo_file)
+    if os.path.exists(os.path.join(args.repo, 'config')):
+        logging.info('Opening repo at ' + args.repo)
+        repo.open()
+    else:
+        logging.info('Creating archive-z2 repo at ' + args.repo)
+        os.makedirs(args.repo)
+        repo.create(OSTree.RepoMode.ARCHIVE_Z2)
+
+
+    # prepare transaction
+
+    repo.prepare_transaction(None)
+    repo.transaction_set_ref(None, ref, commit)
+
+
+    # execute the delta
+
+    flatpak_file = Gio.File.new_for_path(args.flatpak)
+    repo.static_delta_execute_offline(flatpak_file, False, None)
+
+
+    # verify gpg signature
+
+    if args.gpg_homedir:
+        gpg_homedir = Gio.File.new_for_path(args.gpg_homedir)
+    else:
+        gpg_homedir = None
+    if args.keyring:
+        trusted_keyring = Gio.File.new_for_path(args.keyring)
+    else:
+        trusted_keyring = None
+    gpg_verify_result = repo.verify_commit_ext(commit,
+                                               keyringdir=gpg_homedir,
+                                               extra_keyring=trusted_keyring,
+                                               cancellable=None)
+    # TODO: handle signature requirements
+    if gpg_verify_result and gpg_verify_result.count_valid() > 0:
+        logging.info("valid signature found")
+    else:
+        logging.error("no valid signature")
+        exit(1)
+
+
+    # commit the transaction
+
+    repo.commit_transaction(None)
+
+
+    # grab commit root
+
+    (ret, commit_root, commit_checksum) = repo.read_commit(ref, None)
+    if not ret:
+        logging.critical("commit failed")
+        exit(1)
```

Comment:
I think the above will just raise an exception and you wouldn't ever hit this path.